tests/components/enphase_envoy/test_switch.py

```diff
@@ -106,3 +108,106 @@ async def test_switch_operation(
+    assert entity_state.state == STATE_ON
+
+    # test grid status switch operation
+    await hass.services.async_call(
+        SWITCH_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: test_entity},
+        blocking=True,
+    )
+    mock_envoy.disable_charge_from_grid.assert_awaited_once_with()
+    mock_envoy.disable_charge_from_grid.reset_mock()
+
+    await hass.services.async_call(
+        SWITCH_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: test_entity},
+        blocking=True,
+    )
+    mock_envoy.enable_charge_from_grid.assert_awaited_once_with()
+    mock_envoy.enable_charge_from_grid.reset_mock()
+
+    await hass.services.async_call(
+        SWITCH_DOMAIN,
+        SERVICE_TOGGLE,
+        {ATTR_ENTITY_ID: test_entity},
+        blocking=True,
+    )
+    mock_envoy.disable_charge_from_grid.assert_awaited_once_with()
+    mock_envoy.disable_charge_from_grid.reset_mock()
+
+
+@pytest.mark.parametrize(
+    ("mock_envoy"), ["envoy_metered_batt_relay"], indirect=["mock_envoy"]
+)
+async def test_switch_relay_operation(
+    hass: HomeAssistant,
+    mock_envoy: AsyncMock,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test enphase_envoy switch relay entities operation."""
+    with patch("homeassistant.components.enphase_envoy.PLATFORMS", [Platform.SWITCH]):
+        await setup_integration(hass, config_entry)
+
+    entity_base = f"{Platform.SWITCH}."
+
+    for contact_id, dry_contact in mock_envoy.data.dry_contact_settings.items():
+        name = dry_contact.load_name.lower().replace(" ", "_")
+        test_entity = f"{entity_base}{name}"
+        target_value = mock_envoy.data.dry_contact_status[contact_id].status
+        assert (entity_state := hass.states.get(test_entity))
+        assert (
+            entity_state.state == STATE_ON
+            if target_value == STATE_CLOSED
+            else STATE_OFF
+        )
+
+        await hass.services.async_call(
+            SWITCH_DOMAIN,
+            SERVICE_TURN_OFF,
+            {ATTR_ENTITY_ID: test_entity},
+            blocking=True,
+        )
+
+        mock_envoy.open_dry_contact.assert_awaited_once_with(contact_id)
+        mock_envoy.close_dry_contact.assert_not_awaited()
+        mock_envoy.open_dry_contact.reset_mock()
+
+        await hass.services.async_call(
+            SWITCH_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: test_entity},
+            blocking=True,
+        )
+
+        mock_envoy.close_dry_contact.assert_awaited_once_with(contact_id)
+        mock_envoy.open_dry_contact.assert_not_awaited()
+        mock_envoy.close_dry_contact.reset_mock()
+
+        await hass.services.async_call(
+            SWITCH_DOMAIN,
+            SERVICE_TOGGLE,
+            {ATTR_ENTITY_ID: test_entity},
+            blocking=True,
+        )
+
+        assert (
+            mock_envoy.open_dry_contact.await_count
+            if target_value == STATE_CLOSED
```

Comment:
Add parameters for the target value and different call counts etc instead. We don't want any conditions in tests.

Reply:
See PR #122693.